src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/NumericStatement.scala

```diff
@@ -75,7 +75,7 @@ private[querygeneration] object NumericStatement {
 
       case PromotePrecision(child) => convertStatement(child, fields)
 
-      case CheckOverflow(child, t) =>
+      case CheckOverflow(child, t, _) =>
```

Comment:
We may need handle this new argument.

Reply:
Because snowflake doesn't support this option, what we can do is disable push-down for this case. But this is current behavior, so I prefer to keep it. In addition, I will file a Jira to track it, if snowflake supports this feature, we can support it.